torchdynamo/guards.py

```diff
@@ -298,28 +400,28 @@ def CONSTANT_MATCH(self, guard: Guard):
 
     def NN_MODULE(self, guard: Guard):
         self.ID_MATCH(guard)
-        ref = self.arg_ref(guard)
-        val = self.get(guard.name)
-
-        def setup_guard():
-            assert istype(val.training, bool)
-            self.code.append(f"{ref}.training == {val.training}")
 
-        if hasattr(val, "training"):
-            # There are cases where a monkeypatched object has a guard made between __new__ and __init__
-            setup_guard()
+        val = self.get(guard.name)
+        if hasattr(val, "training") and config.guard_nn_modules:
```

Comment:
Just for my curiosity - Why are we checking for training attr here?

Reply:
Good question - context here: https://github.com/pytorch/torchdynamo/pull/322/files